File: Growdamp/mbf_growdamp_basic_fitting.m
function [s, delta, p] = mbf_growdamp_basic_fitting(data, debug)

x_ax = 1:length(data);
s = polyfit(x_ax,log(abs(data)),1);
c = polyval(s,x_ax);
delta = mean(abs(c - log(abs(data)))./c);
temp = unwrap(angle(data)) / (2*pi);
p = polyfit(x_ax,temp,1);
if debug==1
    figure(31)
    plot(x_ax,log(abs(data)),'DisplayName', 'Raw data')
    hold on
    plot(x_ax, c, 'g','DisplayName', 'Fit')
    legend
end %if

File: python_port/Growdamp/mbf_growdamp_basic_fitting.py
import numpy


def mbfGrowdampBasicFitting(data):

    x_ax = range(len(data))
    s = numpy.polynomial.Polynomial.fit(x_ax, numpy.log(abs(data)), 1)
    c = numpy.polyval(s, x_ax)
    delta = numpy.mean(abs(c - numpy.log(abs(data))) / c)
    temp = numpy.unwrap(numpy.angle(data)) / (2 * numpy.pi)
    p = numpy.polynomial.Polynomial.fit(x_ax, temp, 1)
    return s, delta, p
